### DISCUSSION TITLE
How to set GPIO Slew Rate?

### DISCUSSION THREAD

**Discussion by Deadeye5589** [earlephilhower/arduino-pico#1901] [Q&A]
Hello, 

how can I set the slew rate for a given GPIO pin? The default should be slow and I would like to try to improve my signal quality with the high slew rate setting. Is there a function for that or will I have to write directly to a register?

**Comment by earlephilhower** [comment]
Check the SDK documentation.  There should be a convenience function.  Just call it after doing the proper `pinMode` on the GPIO first.

**Comment by Andy2No** [comment]
The current first two results here look relevant:

https://github.com/search?q=repo%3Araspberrypi%2Fpico-sdk%20slew&type=code


> void gpio_set_slew_rate(uint gpio, enum gpio_slew_rate slew)

> enum gpio_slew_rate {
    GPIO_SLEW_RATE_SLOW = 0,  ///< Slew rate limiting enabled
    GPIO_SLEW_RATE_FAST = 1   ///< Slew rate limiting disabled
};

**Comment by Deadeye5589** [comment]
Got it, thanks
